Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerImageAttachmentsView.swift

```diff
@@ -55,39 +50,43 @@ open class _ChatMessageComposerImageAttachmentsView<ExtraData: ExtraDataTypes>:
         collectionView.dataSource = self
         collectionView.delegate = self
         collectionView.register(
-            _ChatMessageComposerImageAttachmentCollectionViewCell<ExtraData>.self,
-            forCellWithReuseIdentifier: _ChatMessageComposerImageAttachmentCollectionViewCell<ExtraData>.reuseId
+            uiConfig.messageComposer.imageAttachmentCollectionViewCell.self,
+            forCellWithReuseIdentifier: uiConfig.messageComposer.imageAttachmentCollectionViewCell.reuseId
         )
     }
     
     override open func setUpLayout() {
-        collectionView.translatesAutoresizingMaskIntoConstraints = false
-        embed(collectionView)
-        
+        flowLayout.scrollDirection = .horizontal
         flowLayout.itemSize = imagePreviewItemSize
         flowLayout.sectionInset = layoutMargins
+
+        embed(collectionView, insets: .init(top: layoutMargins.top, leading: 0, bottom: layoutMargins.bottom, trailing: 0))
+        collectionView.heightAnchor.pin(equalToConstant: imagePreviewItemSize.height).isActive = true
     }
     
     override open func updateContent() {
         collectionView.reloadData()
     }
-
-    // MARK: - UICollectionView
     
     public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        images.count
+        content.count
     }
     
     public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         guard let cell = collectionView
             .dequeueReusableCell(
-                withReuseIdentifier: _ChatMessageComposerImageAttachmentCollectionViewCell<ExtraData>.reuseId,
+                withReuseIdentifier: uiConfig.messageComposer.imageAttachmentCollectionViewCell.reuseId,
                 for: indexPath
             ) as? _ChatMessageComposerImageAttachmentCollectionViewCell<ExtraData>
-        else { return UICollectionViewCell() }
-        
-        cell.imageView.image = images[indexPath.row]
-        cell.discardButtonHandler = { [weak self] in self?.didTapRemoveItemButton?(indexPath.row) }
+        else {
+            return UICollectionViewCell()
+        }
+
+        cell.uiConfig = uiConfig
```

Comment:
nit: Maybe overthinking, but I think it would be nice to put here some comment about `UIConfig` and why it is needed to pass it on `UITableViewCell` / `UICollectionViewCell` instances...

Reply:
I actually thought about creating our own extension to reuse cells, where we can simplify the use of the native `dequeueCell` and in that extension we provide the uiConfig always. Because this is already the 2nd time I lose a lot of time to understand why the uiconfig overriding is not working 😕 So if we have that always done, it will be much safer.